File: market-backend/src/main/java/com/ais/marketbackend/fel/application/services/interfaces/FelService.java
package com.ais.marketbackend.fel.application.services.interfaces;

import com.ais.marketbackend.fel.application.dtos.DocumentoFelResumen;
import com.ais.marketbackend.shared.domain.Pagina;
import java.util.List;

public interface FelService {

    /** Emite el documento FEL de una venta completada — falla si ya tiene uno o la venta no está completada. */
    DocumentoFelResumen emitir(Long tiendaId, Long ventaId);

    DocumentoFelResumen anular(Long tiendaId, Long id, String motivo);

    /** Reintenta la certificación de un documento en estado ERROR. */
    DocumentoFelResumen reintentar(Long tiendaId, Long id);

    DocumentoFelResumen obtener(Long tiendaId, Long id);

    /** Sin paginar — uso interno. El endpoint público usa la variante paginada. */
    List<DocumentoFelResumen> listarPorTienda(Long tiendaId);

    Pagina<DocumentoFelResumen> listarPorTienda(Long tiendaId, int pagina, int tamano);
}

File: market-backoffice/src/services/fel.service.ts
import { apiClient } from '@/services/http/ApiClient'
import { API_ENDPOINTS } from '@/config/endpoints'
import type { DocumentoFel } from '@/types/fel'
import type { Pagina } from '@/types/pagina'

class FelService {
  listarPorTienda(tiendaId: number, pagina: number, tamano: number, signal?: AbortSignal) {
    return apiClient.get<Pagina<DocumentoFel>>(API_ENDPOINTS.fel.porTienda(tiendaId), {
      params: { page: pagina, size: tamano },
      signal,
    })
  }

  obtener(tiendaId: number, id: number, signal?: AbortSignal) {
    return apiClient.get<DocumentoFel>(API_ENDPOINTS.fel.porId(tiendaId, id), { signal })
  }

  emitir(tiendaId: number, ventaId: number) {
    return apiClient.post<DocumentoFel>(API_ENDPOINTS.fel.emitir(tiendaId, ventaId))
  }

  reintentar(tiendaId: number, id: number) {
    return apiClient.post<DocumentoFel>(API_ENDPOINTS.fel.reintentar(tiendaId, id))
  }

  anular(tiendaId: number, id: number, motivo: string) {
    return apiClient.post<DocumentoFel>(API_ENDPOINTS.fel.anular(tiendaId, id), { motivo })
  }
}

export const felService = new FelService()
